lib/tessel/deploy.js

```diff
@@ -9,12 +10,16 @@ var fsTemp = require('fs-temp');
 var glob = require('glob');
 var Ignore = require('fstream-ignore');
 var Reader = require('fstream').Reader;
+var request = require('request');
 var tar = require('tar');
 var uglify = require('uglify-js');
+var urljoin = require('url-join');
+var zlib = require('zlib');
 
 // Internal
 var commands = require('./commands');
 var logs = require('../logs');
+var provision = require('./provision'); // jshint ignore:line
```

Comment:
> That being said, I think it's a really bad pattern to extend stuff defined elsewhere in files.

Agree, and eventually I'd like to get that cleaned up. Maybe after pre-compiled modules is done.